ee/maintained-apps/ingesters/winget/ingester.go

```diff
@@ -166,66 +175,149 @@ func (i *wingetIngester) ingestOne(ctx context.Context, input inputApp) (*mainta
 	}
 
 	var out maintained_apps.FMAManifestApp
+	var selectedInstaller *installer
+	var installScript, uninstallScript string
+	productCode := m.ProductCode
 
-	// TODO: handle non-machine scope (aka .exe installers)
-	var installScript, uninstallScript, installerURL, sha256 string
+	// if we have a provided install script, use that
+	if input.InstallScriptPath != "" {
+		scriptBytes, err := os.ReadFile(input.InstallScriptPath)
+		if err != nil {
+			return nil, ctxerr.Wrap(ctx, err, "reading provided install script file")
+		}
 
-	// Some data is present on the top-level object, so try to grab that first
-	if m.InstallerType == installerTypeMSI || m.Scope == machineScope {
-		installScript = file.GetInstallScript(m.InstallerType)
-		uninstallScript = file.GetUninstallScript(m.InstallerType)
+		installScript = string(scriptBytes)
+	}
+
+	if input.UninstallScriptPath != "" {
+		scriptBytes, err := os.ReadFile(input.UninstallScriptPath)
+		if err != nil {
+			return nil, ctxerr.Wrap(ctx, err, "reading provided uninstall script file")
+		}
+
+		uninstallScript = string(scriptBytes)
 	}
 
-	// Walk through the installers and get any data we missed
 	for _, installer := range m.Installers {
-		if (installer.Scope == machineScope || m.Scope == machineScope || (installer.Scope == "" && m.Scope == "")) &&
-			installer.Architecture == arch64Bit {
-			// Use the first machine scoped installer
-			installerURL = installer.InstallerURL
-			sha256 = installer.InstallerSha256
-			installerType := installer.InstallerType
-			if installerType == "" || installerType == installerTypeWix {
-				// try to get it from the URL
-				// TODO: this may not work in all situations
-				// TODO: also, "wix" might always mean an MSI installer, in which case we should
-				// just use that
-				installerType = strings.Trim(filepath.Ext(installerURL), ".")
-			}
-			if installScript == "" {
-				installScript = file.GetInstallScript(installerType)
-			}
-			if uninstallScript == "" {
-				uninstallScript = file.GetUninstallScript(installerType)
+		installerType := m.InstallerType
+		if installerType == "" || isVendorType(installerType) {
+			installerType = installer.InstallerType
+		}
+
+		if installerType == "" || isVendorType(installerType) {
+			// try to get it from the URL
+			installerType = strings.Trim(filepath.Ext(installer.InstallerURL), ".")
+		}
+
+		scope := m.Scope
+		if scope == "" {
+			scope = installer.Scope
+			if scope == "" {
+				if installerType == installerTypeMSI {
+					scope = machineScope
+				}
 			}
+		}
```

Comment:
can flatten this a bit:

```suggestion
		scope := m.Scope
                if scope == "" {
	           scope = installer.Scope
                }
                if scope == "" && installerType == installerTypeMSI {
	           scope = machineScope
                }

```